### PR TITLE
Update text-style component

### LINKED ISSUE #210: Update text-style component
Diff [here](https://github.com/Shopify/polaris/compare/v2.2.0...v2.11.0#diff-66a7abb8f39a28e40bea8a08b2609fb4R5)

### PR DESCRIPTION
**Pull request by vladucu**
fix #210
Diff [here](https://github.com/Shopify/polaris/compare/v2.2.0...v2.11.0#diff-66a7abb8f39a28e40bea8a08b2609fb4R5)

### CODE REVIEW

File: addon/templates/components/polaris-text-style.hbs [hunk 1]
```diff
@@ -1,5 +1,26 @@
-{{#if hasBlock}}
-  {{yield}}
-{{else}}
-  {{text}}
+{{!-- NOTE: This is a bit hacky, but needed to support dynamically rendering different tag based on variation --}}
```
Comment (vladucu) [inline addon/templates/components/polaris-text-style.hbs]: not sure if there's a simple way to do this

File: addon/components/polaris-text-style.js [hunk 2]
```diff
@@ -40,12 +63,24 @@ export default Component.extend({
    */
   text: null,
 
+  dataTestTextStyle: true,
```
Comment (vladucu) [inline addon/components/polaris-text-style.js]: passing this for tagless component cause passing something like `data-test-*` will result in a warning

File: addon/templates/components/polaris-text-style.hbs (hunk 1 above)
Comment (andrewpye) [reply addon/templates/components/polaris-text-style.hbs]: How about [`wrapper-element`](https://github.com/smile-io/ember-polaris/blob/master/addon/components/wrapper-element.js)? 🤔

Comment (vladucu) [reply addon/templates/components/polaris-text-style.hbs]: tried that, but it won't re-render if `elementTagName` changes

Comment (andrewpye) [reply addon/templates/components/polaris-text-style.hbs]: True... maybe use the approach in [`polaris-list`](https://github.com/smile-io/ember-polaris/blob/master/addon/templates/components/polaris-list.hbs#L2) then?

Comment (andrewpye) [reply addon/templates/components/polaris-text-style.hbs]: Still not perfect, but IMO an improvement 🤷‍♂️

Comment (andrewpye) [reply addon/templates/components/polaris-text-style.hbs]: True... maybe use the approach in [`polaris-list`](https://github.com/smile-io/ember-polaris/blob/master/addon/templates/components/polaris-list.hbs#L2) then? Still not perfect but maybe an improvement 🤷‍♂️

Comment (vladucu) [reply addon/templates/components/polaris-text-style.hbs]: yeah, guess we can use that

File: addon/components/polaris-text-style.js [hunk 3]
```diff
@@ -4,13 +4,36 @@ import { isEmpty } from '@ember/utils';
 import { classify } from '@ember/string';
 import layout from '../templates/components/polaris-text-style';
 
+const variationValue = {
```
[the remark this answers is not in the document]
Comment (vladucu) [reply addon/components/polaris-text-style.js]: true that this is Polaris uses in 2.11, though in 2.12 they use this...so just _traveled in the future_ 😄

**Comment by andrewpye** [comment]
Actually just noticed the failing tests, fix that and we're good 😉

**Pull request merged**